### PR TITLE
net-p2p/p2pool-monero-bin: Added 1.8 ebuild

### PR DESCRIPTION
**Pull request by ethark**
The real ebuild requires some dependencies I couldn't get working. But luckily they also distribute this neat little statically-linked binary. I haven't actually tested this on aarch64, but it should be fairly straightforward.

I've tested it on amd64, and I'm pretty sure it doesn't need multilib support because it's staticially-linked

### CODE REVIEW

**Comment by ethark** [comment]
whoops, forgot a ${PV} there.

File: net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Copyright 1999-2021 Gentoo Authors
```
Comment (00-matt) [inline net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild:1]:
```suggestion
# Copyright 1999-2022 Gentoo Authors
```

File: net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+#To prevent confusion with the bitcoin version of p2pool, this ebuild has been named p2pool-monero
```
Comment (00-matt) [inline net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild:6]: Good idea :)

File: net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+DESCRIPTION="Decentralized pool for Monero mining (Static Binary)"
+HOMEPAGE="https://p2pool.io https://github.com/SChernykh/p2pool"
+SRC_URI="
+	x86?     ( https://github.com/SChernykh/p2pool/releases/download/v${PV}/p2pool-v${PV}-linux-x64.tar.gz )
+	amd64?   ( https://github.com/SChernykh/p2pool/releases/download/v${PV}/p2pool-v${PV}-linux-x64.tar.gz )
+	aarch64? ( https://github.com/SChernykh/p2pool/releases/download/v${PV}/p2pool-v${PV}-linux-aarch64.tar.gz )
+"
```
Comment (00-matt) [inline net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild:13]: The AMD64 binary won't run on x86 :(

```suggestion
SRC_URI="
	amd64?   ( https://github.com/SChernykh/p2pool/releases/download/v${PV}/p2pool-v${PV}-linux-x64.tar.gz )
	aarch64? ( https://github.com/SChernykh/p2pool/releases/download/v${PV}/p2pool-v${PV}-linux-aarch64.tar.gz )
"
```

File: net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+LICENSE="GPL-3+"
```
Comment (00-matt) [inline net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild:15]: We need to add the licenses for all the dependencies that are statically linked in with p2pool :/ - I got these from the `external` directory.

```suggestion
LICENSE="BSD GPL-3+ ISC LGPL-3+ MIT"
```

File: net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+SLOT="0"
+KEYWORDS="~arm64 ~amd64 ~x86"
```
Comment (00-matt) [inline net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild:17]: We can use `-*` to say that architectures not mentioned are unsupported.

```suggestion
KEYWORDS="-* ~amd64 ~arm64 "
```

https://devmanual.gentoo.org/keywording/

**Review by 00-matt — COMMENTED** [review]
Great! Please add the `Manifest file (you can run `pkgdev manifest` or `repoman manifest` to generate this), and a `metadata.xml` file.

File: net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild (hunk 3 above)
Comment (ethark) [reply net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild:13]: Oh that is surprising. I think I must have misread "x64" as "x86". I hate that confusing naming scheme.

File: net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild (hunk 5 above)
Comment (ethark) [reply net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild:17]: Oh, good point. I guess `-*` does make sense for a binary package like this, as the ebuild doesn't do anything for any other arch

File: net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild (hunk 4 above)
Comment (ethark) [reply net-p2p/p2pool-monero-bin/p2pool-monero-bin-1.8.ebuild]: Good catch. Thank you.

**Comment by ethark** [comment]
Alright, I think I've generated the Manifest and squashed the commits correctly. How does that look?

**Review by 00-matt — APPROVED** [review]
Thanks again!

I've replaced `aarch64` with `arm64` and added the `metadata.xml` file.

Also if you want to keep contributing (or to any Gentoo repository) you might like `pkgcheck` and `pkgdev`. `pkgcheck` can scan your changes for common issues, and `pkgdev` can commit for you, making sure to add add things like the [GCO Signed-off-by trailer](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).

**Pull request closed without merging**